### PR TITLE
git 2.42.0

### PR DESCRIPTION
**Pull request by palday**
also bump OpenSSL compat

### CODE REVIEW

File: G/Git/build_tarballs.jl [hunk 1]
```diff
@@ -114,7 +114,7 @@ dependencies = [
     HostBuildDependency("Gettext_jll"),
     Dependency("LibCURL_jll"; compat="7.73.0,8"),
     Dependency("Expat_jll"; compat="2.2.10"),
-    Dependency("OpenSSL_jll"; compat="1.1.10"),
+    Dependency("OpenSSL_jll"; compat="3.0.0"),
```
Comment (giordano) [inline G/Git/build_tarballs.jl]:
```suggestion
    Dependency("OpenSSL_jll"; compat="3.0.8"),
```

**Review by giordano — COMMENTED** [review]
Does the Windows binary lib link to openssl?

**Comment by palday** [comment]
@giordano I have no idea 🙁 

But some other JLLs I use have bumped OpenSSL compat and I saw that Git_jll was the blocker in updating overall because of the OpenSSL dep.

**Comment by giordano** [comment]
Uhm, sounds like they vendor all the deps.

**Pull request merged**